### ISSUE TITLE
NAM

### ISSUE THREAD

**Issue by Ronky123** [Christian-lyc/NAM#3]
Dear Author:
Could you send the complete NAM code to me?Thanks a lot!

**Comment by qq274983722** [comment]
Dear Authors:：
I also need your code to compare the experimental results. My email address is 12021131688@stu.nxu.edu.cn .Thank you!

**Comment by fengzhenfeng** [comment]
Dear Authors:
Could you give me your complete NAM module. My email 2112112242@zjut.edu.cn Thanks!

**Comment by huahuabai** [comment]
Dear Authors:
Could you give me your complete NAM module. My email 1937792073@qq.com Thanks!

**Comment by lscc777** [comment]
Dear Authors:
Could you give me your complete NAM module? My email 2192133522@qq.com Thanks!

**Comment by beam-fly** [comment]
Dear Authors:
Could you give me your complete NAM module? My email [876336921@qq.com]Thanks!

**Comment by xxxjjjx** [comment]
Dear Authors:
Could you give me your complete NAM module? My email [[2532921390@qq.com]Thanks!

**Comment by Azhihong** [comment]
Could you send me your complete NAM code. My email [20201201012@cqu.edu.cn] Thanks a lot!

**Comment by xiaobaitubaibaibai** [comment]
Could you send me your complete NAM code. My email [[2190184760@qq.com]] Thank you!

**Comment by QandB79** [comment]
Dear Authors:
Your work has inspired me a lot！
Could you give me your complete NAM module. My email [2321165795@qq.com] Thanks a lot!

**Comment by kk130629** [comment]
您好 ，可以把MobileNetV2这块代码给我一份吗？我想做对比试验  我的邮箱 1355780613@qq.com  谢谢

**Comment by huahuabai** [comment]
抱歉，暂未收到作者回复


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Christian-lyc/NAM"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月27日(星期天) 中午11:35
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Christian-lyc/NAM] NAM (Issue #3)





 
您好 ，可以把MobileNetV2这块代码给我一份吗？我想做对比试验  我的邮箱 ***@***.***  谢谢

**Comment by shell-filter** [comment]
Dear Authors:
Could you give me your complete NAM code. Thanks you very much! My email [1738393394@qq.com]

**Comment by yahuyang** [comment]
Dear Authors:
Could you give me your complete NAM code. Thanks you very much! My email [[yyh_19940317@163.com](mailto:yyh_19940317@163.com)]

**Comment by zhouwy-lv** [comment]
Dear Authors:
Could you give me the part of MobilenetV2 combined with attention? Thank you very much! My email [545641005@qq.com]

**Comment by tuziqq** [comment]
Dear Authors:
Could you give me your complete NAM code. Thanks you very much! My email [[1294574728@163.com]

**Comment by xiaobai111111223** [comment]
Dear Authors:
Could you give me your complete NAM code. Thanks you very much! My email [1319586335@qq.com]

**Comment by nisawan** [comment]
Dear Authors:
Could you give me your complete NAM code. Thank you very much! My email [nisawan.live@gmail.com]

**Comment by Gchang9** [comment]
Dear Authors:：
I also need your code to compare the experimental results.  Thank you very much!My email address is [xidianlichang@163.com]

**Comment by xiaobaitubaibaibai** [comment]
不好意思，并没有代码




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年9月14日(星期三) 晚上8:18
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [Christian-lyc/NAM] NAM (Issue #3)





 
Dear Authors:：
 I also need your code to compare the experimental results.  Thank you very much!My email address is ***@***.***

**Comment by CesareDavidePace** [comment]
Dear Authors:
Could you give me your complete NAM code. Thank you very much! My email cesaredavide.pace@studentmail.unicas.it

**Comment by lucky-asia** [comment]
Dear Authors:
Could you give me your complete NAM module. My email [2323206675@qq.com] Thanks!

**Comment by 23jisuper** [comment]
作者 您好！关于空间注意力模块能够提供一下源代码吗 ？  非常感谢，2043567695@qq.com

**Comment by lucky-asia** [comment]
import torch
from torch import nn



class wca(nn.Module):

    def __init__(self, channel, k_size=3):
        super(wca, self).__init__()
        self.avg_pool = nn.AdaptiveAvgPool2d(1)
        self.conv = nn.Conv1d(1, 1, kernel_size=k_size, padding=(k_size - 1) // 2, bias=False)
        self.bn2 = nn.BatchNorm2d(channel)
        self.sigmoid = nn.Sigmoid()

    def forward(self, x):

        y = self.avg_pool(x)
        y = self.bn2(y)
        weight = self.bn2.weight.data.abs()
        weights = 0.01 * weight
        y = torch.mul(weights, x)
        y = self.sigmoid(y)

        return x * y.expand_as(x)






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Christian-lyc/NAM"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年12月4日(星期一) 晚上6:53
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Christian-lyc/NAM] NAM (Issue #3)





 
作者 您好！关于通道注意力模块能够提供一下源代码吗 ？  ***@***.***
